pkg/api/v1/types.go

```diff
@@ -302,6 +302,7 @@ type SyncEvent string
 const (
 	Created      SyncEvent = "Created"
 	Deleted      SyncEvent = "Deleted"
+	PresetFailed SyncEvent = "PresetFailed"
```

Comment:
FieldPresetFailed

Reply:
What's the rationale behind naming this event `FieldPresetFailed`?

Reply:
Right now the PresetFailed Event seems to be used for any kind of error. I think that @fabiand is right that a conflict deserves an extra event type. Consider "conflict", like it is used on presets. Taking the example from https://kubernetes.io/docs/tasks/inject-data-application/podpreset/#conflict-example:

```
Events:
  FirstSeen             LastSeen            Count   From                    SubobjectPath               Reason      Message
  Tue, 07 Feb 2017 16:56:12 -0700   Tue, 07 Feb 2017 16:56:12 -0700 1   {podpreset.admission.kubernetes.io/podpreset-allow-database }    conflict  Conflict on pod preset. Duplicate mountPath /cache.
```

Also consider using the `SubobjectPath`, to have the reference to the preset in the event, since the event itself needs to be attached to the vm.